src/Maths/Silk.NET.Intrinsics/Avx/AvxRegister.Double.cs

```diff
@@ -0,0 +1,330 @@
+// This file is part of Silk.NET.
+// 
+// You may modify and distribute Silk.NET under the terms
+// of the MIT license. See the LICENSE file for details.
+
+namespace Silk.NET.Intrinsics.Avx
+{
+    public partial struct AvxRegister : IRegister<double>
```

Comment:
What does "Register" mean?

Reply:
The name is poorly chosen, but in the context of the intrinsics library a register is a class capable of performing mathematical operations for a given type using a SIMD register such as AVX or SSE.

Reply:
Maybe `IVector` or something would be a better name?

Is `Avx` referring to `VEX` encoded (128-bit or 256-bit) vectors or strictly 256-bit vectors? If the latter, it might also benefit from a clearer name.

Reply:
Yeah VEX-encoded, but I don't want that leaking out to the user as I want this to be relatively easy to use and not try to introduce too many new concepts to the user - all the user needs to know is "fast maths ooh shiney"

Reply:
Is it being VEX encoded an important detail? The operation is ultimately the same, just with better codegen under VEX.
It normally only gets interesting when the size changes, since that changes how much you are processing, etc.